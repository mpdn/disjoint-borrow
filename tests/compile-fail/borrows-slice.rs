extern crate disjoint_borrow;
use disjoint_borrow::DisjointSlice;

fn main() {
    let mut a = [1, 2, 3];
    let mut sl = DisjointSlice::new(&mut a);
    a[0] = 1;
    sl.get(0..1);
}