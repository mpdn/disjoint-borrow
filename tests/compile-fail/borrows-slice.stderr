error[E0503]: cannot use `a` because it was mutably borrowed
 --> $DIR/borrows-slice.rs:7:5
  |
6 |     let mut sl = DisjointSlice::new(&mut a);
  |                                     ------ borrow of `a` occurs here
7 |     a[0] = 1;
  |     ^^^^ use of borrowed `a`
8 |     sl.get(0..1);
  |     -- borrow later used here

error[E0506]: cannot assign to `a[_]` because it is borrowed
 --> $DIR/borrows-slice.rs:7:5
  |
6 |     let mut sl = DisjointSlice::new(&mut a);
  |                                     ------ borrow of `a[_]` occurs here
7 |     a[0] = 1;
  |     ^^^^^^^^ assignment to borrowed `a[_]` occurs here
8 |     sl.get(0..1);
  |     -- borrow later used here

Some errors have detailed explanations: E0503, E0506.
For more information about an error, try `rustc --explain E0503`.
